Formula/hdf5@1.8.rb

```diff
@@ -26,6 +26,8 @@ class Hdf5AT18 < Formula
   depends_on "gcc" # for gfortran
   depends_on "szip"
 
+  uses_from_macos "m4"
```

Comment:
```suggestion
  uses_from_macos "m4" => :build
```
Given your title?

Reply:
Though given #73797 this may as well be a `depends_on`, possibly.

Reply:
Yes, you are right, I forgot to copy-paste it. Should be fixed now.